### PR TITLE
Implementation of the contact and details page

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
Development of the page of the list of contacts and the page of details of the selected contact.

Video:
https://user-images.githubusercontent.com/84338513/163871237-0bf598b7-07a9-4e38-ad7b-94f70af065b6.mov

Test:
Screen Shot 2022-04-18 at 17 16 37

### CODE REVIEW

**Review by FranciscoGaraventa — APPROVED** [review]

**Review by LeonardoMolleker — APPROVED** [review]

File: lib/model/name.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    return Name(
+      title: json['title'],
+      first: json['first'],
+      last: json['last'],
+    );
+  }
+
+  late String title;
+  late String first;
+  late String last;
```
Comment (MarcosCGlobant) [inline lib/model/name.dart]: why did you defined this values as late?

File: lib/model/location.dart [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      coordinates: Coordinates.fromJson(json['coordinates']),
+      timeZone: TimeZone.fromJson(json['timezone']),
+    );
+  }
+
+  Street street;
+  String city;
+  String state;
+  String country;
+  var postcode;
```
Comment (MarcosCGlobant) [inline lib/model/location.dart:32]: why "postcode" is a var?

**Review by Juliana-Dimeglio — APPROVED** [review]

File: lib/model/name.dart (hunk 1 above)
Comment (ezequielsetzesglobant) [reply lib/model/name.dart:18]: because it throws an error if I don't define it as late

File: lib/model/location.dart (hunk 2 above)
Comment (ezequielsetzesglobant) [reply lib/model/location.dart:32]: because the postcode is sometimes a String and sometimes it's an int

File: lib/model/name.dart (hunk 1 above)
Comment (ezequielsetzesglobant) [reply lib/model/name.dart]: It was from an implementation that I was doing previously in the models and I needed to change it in that file
now i change it

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Pull request merged**